### DISCUSSION TITLE
Access over Local Area Network

### DISCUSSION THREAD

**Discussion by nakumah** [ninenines/cowboy#1622] [Q&A] [answered]
Hello. 

I do not know if this has already been discussed. If it has please point me to the right places

I am trying to access a cowboy server on one computer(A) from another computer (B) over a local area network (my wifi). How can I do this?

The connection between the computers is working fine. I tested accessing a local HTML webpage on one computer from the other and it works well. When i try to access the cowboy server it doesn't work. I get HTTP Error Code 400. 

The way I am accessing is as follows: in the browser on `Computer B` type in the IP address of `computer A` and its port number `(192.168.1.69:8080)`.

Please help.

**Comment by essen** [comment]
Please open dev tools in the browser and you will get the headers sent for the request and response that you can paste here. The dispatch list would also help.

**Comment by nakumah** [comment]
### Snip of browser network dev tools
![image](https://github.com/ninenines/cowboy/assets/65168520/4641c150-8129-4994-b75b-b048f0884231)
There is no response

### Cowboy listener - xxx_app.erl
```

start(_StartType, _StartArgs) ->
    Dispatch = cowboy_router:compile([
        {
            <<"localhost">>,
            [
                { <<"/">>, hello_handler, [] },  %%  returns text
                { <<"/reduce">>, reduce_handler,[]},  %% returns json data
                { <<"/scan">>, scan_handler,[]} %% returns json data
            ]
        } 
    ]),
    {ok, _} = cowboy:start_clear(
        listener,
        [{port, 8080}],
        #{env => #{dispatch => Dispatch}}
    ),
    vc418_server_sup:start_link().

```

**Reply by essen** [reply, marked as answer]
That's because you specify `<<"localhost">>` as the host in your dispatch list. Doing this means only requests that have `Host: localhost` (with/without port) will succeed.

To make it work you can accept any host by changing `<<"localhost">>` to `'_'`.

**Reply by nakumah** [reply]
Thank you very much. Let me try this.
